diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/BlockStructureBraces.kt

```diff
@@ -180,12 +175,15 @@ class BlockStructureBraces : Rule("block-structure") {
     private fun checkCloseBrace(node: ASTNode, configuration: BlockStructureBracesConfiguration) {
         if (!configuration.closeBrace) return
         val space = node.findChildByType(RBRACE)!!.treePrev
-        if (space == null || space.elementType != WHITE_SPACE || !space.text.contains("\n"))
+        if (checkBraceNode(space))
             BRACES_BLOCK_STRUCTURE_ERROR.warnAndFix(configRules, emitWarn, isFixMode, "no newline before closing brace",
                     (space.treeNext ?: node.findChildByType(RBRACE))!!.startOffset) {
             }
     }
 
+    private fun checkBraceNode(node: ASTNode, isContains: Boolean = false) =
+            (node.elementType != WHITE_SPACE || !isContains.xor(node.text.contains("\n")))
```

Comment:
```suggestion
    private fun checkBraceNode(node: ASTNode, shouldContainNewline: Boolean = true) =
            (node.elementType != WHITE_SPACE || shouldContainNewline  == node.text.contains("\n"))
```
If I understood this method's idea correctly) Try to avoid overcomplications (negative xor seems like one :) ) and make variable names more self-descriptive.